examples/chart/teleport-kube-agent/templates/statefulset.yaml

```diff
@@ -13,6 +13,10 @@ metadata:
   {{- if .Values.extraLabels.deployment }}
     {{- toYaml .Values.extraLabels.deployment | nindent 4 }}
   {{- end }}
+  {{- if .Values.annotations.deployment }}
+  annotations:
+    {{- toYaml .Values.annotations.deployment | nindent 4 }}
```

Comment:
can we use separate annotations for the statefulset?
the deployment is a legacy thing we are trying to get rid of

Reply:
Also, our documentation for the values already states that it will be applied to the StatefulSet, so if anything, we have a bug today in that it is not.

```
  # annotations.deployment(object) -- contains the Kubernetes annotations
  # put on the `Deployment` or `StatefulSet` resource created by the chart.
  deployment: {}
```

Reply:
Then it's my fault